### PR TITLE
[FIX] stock_delivery: memory error

### PR DESCRIPTION
**Pull request by asno-odoo**
compute_packages method fetches all columns for stock_move_lines and its result_package_id. If stock_move_lines are a big number, it causes memoryerror. Pre fetch only the required fields. Also its better to use _read_group then search count in a loop.

```
select COUNT(*) from stock_move_line
+---------+
| count   |
|---------|
| 2546604 |
+---------+
```

While fetching 4000 records.
time before: 4.77s
time after: 3.50s

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Sehnde — CHANGES_REQUESTED** [review]
Could you edit the commit message to indicate how SMLs are 'needed' to crash? 
Also since it's a perf issue (with error but still perf), could you add a table showing the difference in perf before and after your fix?
Here's an example: odoo/enterprise#56167

Apart from that, good job

**Comment by asno-odoo** [comment]
> Could you edit the commit message to indicate how many SMLs are 'needed' to crash? Also since it's a perf issue (with error but still perf), could you add a table showing the difference in perf before and after your fix? Here's an example: [odoo/enterprise#56167](https://github.com/odoo/enterprise/pull/56167)
> 
> Apart from that, good job

Hey Stephane,
its done.

**Review by Sehnde — APPROVED** [review]

**Comment by asno-odoo** [comment]
> @asno-odoo good enough. Now you need to correct your code so that it passes style checks and it should be ok.

done

File: addons/stock_delivery/models/stock_picking.py [hunk 1]
```diff
@@ -34,9 +34,18 @@ def _auto_init(self):
...
+              ('result_package_id', '!=', False)],
+              groupby=['picking_id'],
+              aggregates=['__count'],
+        ))
+        self.fetch(['move_line_ids'])
+        self.move_line_ids.fetch(['result_package_id'])
         for package in self:
             packs = set()
-            if self.env['stock.move.line'].search_count([('picking_id', '=', package.id), ('result_package_id', '!=', False)]):
+            if counts.get(package, 0):
```
[the remark this answers is not in the document]
Comment (asno-odoo) [reply addons/stock_delivery/models/stock_picking.py]: thanks its done. It wasn't related to any task but solving more generic/standard issues from rd-upgrade team. The source of the problem are some databases sent with large amount of stock_move_line records. saas-16.4 is no longer a valid target for upgrade so making this in 17.0 makes more sense.

[the remark this answers is not in the document]
Comment (asno-odoo) [reply addons/stock_delivery/models/stock_picking.py]: sorry for not seeing the message earlier but changes are done.

**Review by clesgow — APPROVED** [review]
Thanks!

@robodoo r+

**Pull request closed without merging**